### ISSUE TITLE
Google login for extensions is blocked on brave-core

### ISSUE THREAD

**Issue by jumde** [brave/brave-browser#14041]
## Description 
Google login for extensions is blocked for extensions. 

## Steps to Reproduce

   1. Enable Google Login for extensions in brave://settings/extensions
   2. Download Google Keep extension and try to login to Google.
   3. Login on browser works, but extension does not log-in. 

## Actual result:
Google login doesn't work for extensions

## Expected result:
Extension should login as well.

## Reproduces how often: 
Easily

## Brave version (brave://version info)
1.21.36 Chromium: 88.0.4324.96 (Official Build) nightly (x86_64)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by agrawalsourav98** [comment]
I am also facing the same issue, it keeps taking me to the google login screen although I am already logged in

**Comment by jocon-grmtx** [comment]
Same here.  I make extensive use of Google keep, to a point where I have to switch to Chrome (I would prefer to stay on Brave !)

**Comment by jumde** [comment]
@agrawalsourav98 @jocon-grmtx - Thanks for the feedback. We are currently investigating this issue, will post updates as we have them. Could you please confirm if you're seeing this message in `brave://signin-internals`: `Missing authorization code due to OAuth outage in Dice` - Thanks!

**Comment by agrawalsourav98** [comment]
Yes, I am attaching the screenshot of the page
![brave-debug](https://user-images.githubusercontent.com/35604369/107610904-c0160800-6c68-11eb-94e1-c056ea5e0bee.png)

**Comment by jumde** [comment]
Thanks for the update @agrawalsourav98

**Comment by r5464y75** [comment]
Was not working for me, however I've been able to get it working by:

- ensuring Keep extension is uninstalled
- enabling Allow Google login for extensions
- relaunching Brave
- reinstalling Keep extension
- clicking on the Keep "sign-in" button
- signing into Google via the newly-opened tab

![2021-02-25 10_53_55-Window](https://user-images.githubusercontent.com/19987916/109086920-31b98000-7758-11eb-816e-82787f80e372.png)

![2021-02-25 10_57_29-Window](https://user-images.githubusercontent.com/19987916/109086972-4b5ac780-7758-11eb-982e-93c0e16f0b18.png)

**Comment by agrawalsourav98** [comment]
> Was not working for me, however I've been able to get it working by:
> 
> * ensuring Keep extension is uninstalled
> * enabling Allow Google login for extensions
> * relaunching Brave
> * reinstalling Keep extension
> * clicking on the Keep "sign-in" button
> * signing into Google via the newly-opened tab
> 
> Evidence of desired behaviour:
> 
> ![2021-02-25 10_53_55-Window](https://user-images.githubusercontent.com/19987916/109086920-31b98000-7758-11eb-816e-82787f80e372.png)
> 
> Brave build:
> 
> ![2021-02-25 10_57_29-Window](https://user-images.githubusercontent.com/19987916/109086972-4b5ac780-7758-11eb-982e-93c0e16f0b18.png)

It didn't work for me

**Comment by jumde** [comment]
Hi all, just an update here - usage of Google oauth credentials is capped for chromium derivatives. As an alternative, Google is working on an update to `chrome.identity` that will work for chromium derivatives. Hoping this lands upstream soon: https://chromium-review.googlesource.com/c/chromium/src/+/2704571

**Comment by mastino21** [comment]
I still face this login problem in Keep extension and in Drive too (Save to Drive extension).

**Comment by nickgroverx** [comment]
I did these two and it worked for me:
1. Allow google login for extensions, and 
2. Allow all cookies.

**Comment by Excess** [comment]
Still not working as of May 27th, Brave _Version 1.25.68 Chromium: 91.0.4472.77 (Official Build) (64-bit)_

- **Allow Google login for extensions** is _ON_
- Brave's **SHIELDS** are completely _DISABLED_ (no blocking of trackers, cookies, scripts whatsoever)
- Extension that can't login in is listed as allowed on _myaccount.google.com/permissions_

I hope Google uncaps Google oauth credentials for Chromium soon, this is the only thing stopping me from fully migrating to Brave at work too.

**Comment by mastino21** [comment]
It works for me. Version 1.24.86 Chromium: 90.0.4430.212 (Build ufficiale) (a 64 bit) for Linux (Debian).

**Comment by mastino21** [comment]
New update, actually something strange happens. I use multiple Brave profiles with multiple google accounts. As I wrote a few days ago, in one profile everything works (save to Google keep and google drive). In another Brave profile -with exactly the same browser settings- no way, I can't sign-in at all to the extensions: when I sign in I get redirected to google.com and I am signed in but if I click the extension button I am not actually signed in there.

**Comment by jwashburn** [comment]
I'm having this issue too. Keep extension does nothing when I click "Sign in." Tried all the steps above and it still doesn't work. Following this issue now.

**Comment by bsclifton** [comment]
Closing as duplicate of https://github.com/brave/brave-browser/issues/15754
